Furball.Vixie/Graphics/Backends/Veldrid/FullScreenQuad.cs

```diff
@@ -0,0 +1,111 @@
+
+    public class FullScreenQuad {
+        private readonly VeldridBackend _backend;
+
+        private static ushort[] s_quadIndices = new ushort[] {
+            0, 1, 2, 0, 2, 3
+        };
+        
+        private Pipeline     _pipeline;
+        private DeviceBuffer _ib;
+        private DeviceBuffer _vb;
+        public  bool         UseTintedTexture { get; set; }
+
+        public FullScreenQuad(VeldridBackend backend) {
+            this._backend = backend;
+
+            var factory = backend.ResourceFactory;
+            var gd      = backend.GraphicsDevice;
+            var cl      = backend.CommandList;
+            
+            ResourceLayout resourceLayout = backend.ResourceFactory.CreateResourceLayout(new ResourceLayoutDescription(new ResourceLayoutElementDescription("SourceTexture", ResourceKind.TextureReadOnly, ShaderStages.Fragment), new ResourceLayoutElementDescription("SourceSampler", ResourceKind.Sampler, ShaderStages.Fragment)));
+
+            Shader[] shaders = this._backend.ResourceFactory.CreateFromSpirv(new ShaderDescription(ShaderStages.Vertex, Encoding.UTF8.GetBytes(FullScreenQuadShaders.VertexShader), "main"), new ShaderDescription(ShaderStages.Fragment, Encoding.UTF8.GetBytes(FullScreenQuadShaders.FragmentShader), "main"));
+
+            GraphicsPipelineDescription pd = new GraphicsPipelineDescription(new BlendStateDescription(RgbaFloat.Black, BlendAttachmentDescription.OverrideBlend),
+                                                                             DepthStencilStateDescription.Disabled,
+                                                                             new RasterizerStateDescription(FaceCullMode.Back, PolygonFillMode.Solid, FrontFace.Clockwise, true, false),
+                                                                             PrimitiveTopology.TriangleList,
+                                                                             new ShaderSetDescription(new[] {
+                                                                                                          new VertexLayoutDescription(new VertexElementDescription("Position", VertexElementSemantic.TextureCoordinate, VertexElementFormat.Float2), new VertexElementDescription("TexCoords", VertexElementSemantic.TextureCoordinate, VertexElementFormat.Float2))
+                                                                                                      },
+                                                                                                      shaders),
+                                                                             new ResourceLayout[] {
+                                                                                 resourceLayout
+                                                                             },
+                                                                             gd.SwapchainFramebuffer.OutputDescription);
```

Comment:
this looks like a war crime

Reply:
yeah this formatting got FUCKED, ill try to manually fix